src/dataframe/dataframe.jl

```diff
@@ -278,6 +278,26 @@ _columns(df::DataFrame) = getfield(df, :columns)
 nrow(df::DataFrame) = ncol(df) > 0 ? length(_columns(df)[1])::Int : 0
 ncol(df::DataFrame) = length(index(df))
 
+##############################################################################
+##
+## DataFrame consistency check
+##
+##############################################################################
+
+function _check_consistency(df::DataFrame)
+    cols, idx = _columns(df), index(df)
+    ncols = length(cols)
+    @assert length(idx.names) == length(idx.lookup) == ncols "Index and columns of a data frame do not match"
+    ncols == 0 && return nothing
+    nrows = length(cols[1])
+    for i in 2:length(cols)
+        @assert length(cols[i]) == nrows "Length of column $i does not match length of column 1"
```

Comment:
Print column name, its length, and mention that it's likely due to resizing the column vector? Since we expect this kind of error to happen due to user errors rather than bugs in our code, maybe this should be an `ArgumentError`? Not sure.

Reply:
message fixed. I would print `@assert` as this indicates "broken contract". I was considering `ArgumentError`, but I would reserve it for cases when user makes an error when calling a function (not - passes "broken" data).

Reply:
So are you OK with `@assert` or we switch it to `ArgumentError` (I prefer `@assert`, but this is not a crucial thing for me to keep)

Reply:
Yes that's fine.